etna/articles/models.py

```diff
@@ -226,3 +229,166 @@ def latest_items(self) -> Tuple["ArticlePage"]:
+
+    date_text = models.CharField(
+        verbose_name=_("date text"),
+        max_length=100,
+        help_text=_("Date(s) related to the record (max. character length: 100)"),
+    )
+
+    about = RichTextField(verbose_name=_("why this record matters"))
+
+    image_library_link = models.URLField(
+        blank=True,
+        verbose_name="Image library link",
+        help_text="Link to an external image library",
+    )
+
+    print_on_demand_link = models.URLField(
+        blank=True,
+        verbose_name="Print on demand link",
+        help_text="Link to an external print on demand service",
+    )
+
+    featured_article = models.ForeignKey(
+        "articles.ArticlePage",
+        blank=True,
+        null=True,
+        on_delete=models.SET_NULL,
+        verbose_name=_("featured article"),
+    )
+
+    class Meta:
+        verbose_name = "Record Revealed article"
+        verbose_name_plural = "Record Revealed articles"
+
+    content_panels = BasePage.content_panels + [
+        FieldPanel("standfirst"),
+        MultiFieldPanel(
+            heading="Content Warning Options",
+            classname="collapsible collapsed",
+            children=[
+                FieldPanel("display_content_warning"),
+                FieldPanel("custom_warning_text"),
+            ],
+        ),
+        InlinePanel(
+            "gallery_images",
+            heading="Image Gallery",
+            label="Item",
+            min_num=1,
+            max_num=6,
+        ),
+        MultiFieldPanel(
+            heading="Body",
+            children=[
+                FieldPanel("record"),
+                FieldPanel("date_text"),
+                FieldPanel("about"),
+                FieldPanel("image_library_link"),
+                FieldPanel("print_on_demand_link"),
+            ],
+        ),
+        FieldPanel("featured_article"),
+        TopicalPageMixin.get_time_periods_inlinepanel(),
+        TopicalPageMixin.get_topics_inlinepanel(),
+    ]
+
+    promote_panels = MetadataPageMixin.promote_panels + TeaserImageMixin.promote_panels
+
+    @cached_property
+    def gallery_items(self):
+        """
+        Used to access the page's 'gallery_images' for output. Makes use of
+        Django's select_related() and prefetch_related() to efficiently
+        prefetch image and rendition data from the database.
+        """
+        return (
+            self.gallery_images.all()
+            .select_related("image")
+            .prefetch_related("image__renditions")
+        )
+
+
+class PageGalleryImage(Orderable):
+    page = ParentalKey(Page, on_delete=models.CASCADE, related_name="gallery_images")
+    image = models.ForeignKey(
+        "wagtailimages.Image", on_delete=models.SET_NULL, null=True, related_name="+"
+    )
+    is_sensitive = models.BooleanField(
+        verbose_name="Is this image sensitive?",
+        default=False,
+    )
+    alt_text = models.CharField(
```

Comment:
Suggestion: This should not be blank, alt-text is a requirement for screen-reader / assisted viewing. From memory, we cannot fallback to an img model alt-text so it has to be supplied here. Also, from an accessibility point of view, the alt-text also provides context relevant to where an image is shown so it should be tailored here to the image gallery.

Reply:
Good shout!

Reply:
Looks like `alt_text` is already required, so nothing to do here :)